src/frequenz/sdk/config/_config_managing.py

```
@@ -130,6 +163,12 @@ def _read_config(self) -> abc.Mapping[str, Any]:
             except ValueError as err:
                 _logger.error("%s: Can't read config file, err: %s", self, err)
                 error_count += 1
+            except FileNotFoundError:
+                # It is ok for config file to don't exist.
+                _logger.info(
+                    "%s: Config file %s not found. Ignoring it.", self, config_path
+                )
+                error_count += 1
```

Mmm, interesting. Why not an error? I also wonder if we shouldn't do this actually for any exception raised while reading, parsing and validating the config files. The goal of this is to be as resilient as possible in case of a partial failure (we can only read part of the config. And it is really tricky, sometimes I think if one config file read failed, we should probably skip the update completely for all files, as we might get into an inconsistent config.

I'm starting to feel maybe it would be better to restart the actor if there is an issue reading **any** of the requested files (i.e. requiring users to create at least and empty config for the files that are requested in the constructor).

But maybe this is a topic for a different PR. But I would make this a `_logger.error()` in this PR. And maybe catch all exceptions, not just `ValueError` and `FileNotFound` (at least all `OSError`, not just `FileNotFound` as well as any errors that could be railed by `tomlib.load()`, maybe it only raises `ValueError`).

---

Currently it works like this:
Every time any config file change - we read all config files. But not all config files might exists and this cover this case.

Luca, this PR was to fix existing bugs, not redesign ConfigManger. I guess it needs redesign and rethinking but maybe in separate PR and after we agree how it should work.

---

So I log info, if any config file is doesn't exist. 
And (code later) raises ValueError if no config file exist.

---

> Luca, this PR was to fix existing bugs, not redesign ConfigManger. I guess it needs redesign and rethinking but maybe in separate PR and after we agree how it should work.

Yes, 100% agreed, this is also why I said re-evaluating what to do on a failure to read a config file is for another PR.

But catching other types of errors is still a bug that IMHO is in scope for this PR. Since you are catching a new `FileNotFound` why not just catching all `OSError`s?

> So here I log info, if any config file is doesn't exist.
> And (code 3 lines below) raises ValueError if no config file exist. (at least one must exist)

Yeah, but that's only if we can't read **any** config file, if only one failed, no `ValueError` will be raised, and the missing file won't be noticed if we are only monitoring WARNING and up. Also it is more consistent to how we handle the TOML parsing `ValueError`. We log an error if TOML parsing fails. I don't understand why the inconsistency of making one issue emit an error log and the new one just a info.

---

* https://github.com/frequenz-floss/frequenz-sdk-python/issues/1120

---

Ok the whole discussion was because I had impression that having not all config files is not an error.
But it looks like it is unexpected so as you suggested I changed severity of this logging to  error. :) 
Sorry for my missunderstanding

---

No it is OK, originally it was thought like that, only one config file should be enough, but I think that's assuming too much, and is hard to know if missing files are errors or expected, so I opened the new issue to change that in the future.

---

Oh, I see in the last commit you are changing this. I would not address the new issue (#1120) in this PR, I think it needs a bit more thought and it is not strictly related to the bug fixes in this PR.